### DISCUSSION TITLE
An issue while collecting statics to be uploaded on Azure storage

### DISCUSSION THREAD

**Discussion by azeemshah27** [cookiecutter/cookiecutter-django#5374] [General]
I have a project setup with docker, no whitenoise so statics and media will both be on Azure storage 

while creating a production build with `docker compose -f .\docker-compose.production.yml build`

I am running into the following issue 

```shell
django-1  | PostgreSQL is available
django-1  | Traceback (most recent call last):
django-1  |   File "/app/manage.py", line 32, in <module>
django-1  |     execute_from_command_line(sys.argv)
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
django-1  |     utility.execute()
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/management/__init__.py", line 436, in execute
django-1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/management/base.py", line 413, in run_from_argv
django-1  |     self.execute(*args, **cmd_options)
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/management/base.py", line 459, in execute
django-1  |     output = self.handle(*args, **options)
django-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/collectfasta/management/commands/collectstatic.py", line 128, in handle
django-1  |     ret = super().handle(**options)
django-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 171, in handle
django-1  |     self.set_options(**options)
django-1  |   File "/usr/local/lib/python3.12/site-packages/collectfasta/management/commands/collectstatic.py", line 78, in set_options
django-1  |     self.strategy = self._load_strategy()(self.storage)
django-1  |                     ^^^^^^^^^^^^^^^^^^^^^
django-1  |   File "/usr/local/lib/python3.12/site-packages/collectfasta/management/commands/collectstatic.py", line 56, in _load_strategy
django-1  |     raise ImproperlyConfigured(
django-1  | django.core.exceptions.ImproperlyConfigured: No strategy configured, please make sure COLLECTFASTA_STRATEGY is set.
django-1 exited with code 1
``` 

With `COLLECTFASTA_ENABLED = False` in settings, I am able to build the image and statics are being uploaded properly as well 

Should I keep COLLECTFASTA disabled? as I was not able to find any `COLLECTFASTA_STRATEGY` for Azure also I am not very clear on what it will do for me as I don't have a lot of statics right now but it can change in the future

While initiating the project the selections were 

`use_docker` > `yes`
`cloud_provider` > `Azure`
`use_whitenoise` > `no`

It seems like [Collectfasta](https://github.com/jasongi/collectfasta) should not have been installed or may be I am missing something any help will be appreciated, Thanks :)

**Comment by browniebroke** [comment]
Right, collectfasta doesn't support Azure storage so it should not be enabled, nevermind installed.

You can leave it disabled, if speed becomes a problem, you might want to look again if support has been added or suggest it as feature request to collectfasta.

In the meantime, we should fix the template to not install collectfasta when Azure is chosen. If you want to send us a pull request, that would be great.

Thanks for the report!

**Reply by azeemshah27** [reply]
Thanks a lot. I have disabled COLLECTFASTA for now and will revisit it later. I will try to create a pull request as well.

I also had to change `"file_overwrite": False` to `"overwrite_files": False` in 
```python
"default": {
        "BACKEND": "storages.backends.azure_storage.AzureStorage",
        "OPTIONS": {
            "location": "media",
            "overwrite_files": False,
        },
    },
``` 
Otherwise, I was getting the `django.core.exceptions.ImproperlyConfigured: Invalid setting 'file_overwrite' for AzureStorage` error. I'm not sure if this deserves a separate thread, so I'm putting it here.

Another thing I noticed was the presence of Nginx in `compose/production/nginx` 
```
server {
  listen       80;
  server_name  localhost;
  location /media/ {
    alias /usr/share/nginx/media/;
  }
}
```
Should this exist? Azure was selected as the cloud provider, and I don't think it's being referenced anywhere.

**Reply by browniebroke** [reply]
That's great, thanks for letting us know about all these, they do all seem like bugs with Azure storage. They can all be fixed at once if we can.

**Reply by browniebroke** [reply]
Finaly came around to fixing these issues: https://github.com/cookiecutter/cookiecutter-django/pull/5476
